Install drake lcmtypes

fbudin69500 opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/6477)
<!-- Reviewable:end -->

jamiesnape commented:
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/lcmtypes/BUILD, line 354 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnpJz_0z29clMWt72Ux:-KnpJz_0z29clMWt72Uy:buqcco1) ([raw file](https://github.com/robotlocomotion/drake/blob/b2add5d04af858768b5f6ca3c96ad1141eab27d1/drake/lcmtypes/BUILD#L354)):*
> ```Python
>         ":iiwa",
>         ":schunk",
>         ":acrobot_java",
> ```

Make the list of java lcmtypes a variable and use it in the rule above. Also alphabetize.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Really the Java LCM type jars need renaming from `????_java.jar` to `lcmtypes_????.jar`, but this is probably fine for now.

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-KnpM88hjtBpzg7jTkm1:b-31lwep)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
The new commit installs and renames the jars.

---

Review status: 1 of 3 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-Knq6uvgKY6S3qLuHXmR:b-dcns8w)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 1 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/lcmtypes/BUILD, line 354 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnpJz_0z29clMWt72Ux:-Knq75ATxs-Q6lJC1LoG:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/b2add5d04af858768b5f6ca3c96ad1141eab27d1/drake/lcmtypes/BUILD#L354)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Make the list of java lcmtypes a variable and use it in the rule above. Also alphabetize.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
I approve of the general idea, but I'd like to take a closer look at the implementation details tomorrow.

---

Reviewed 1 of 2 files at r1.
Review status: 1 of 3 files reviewed at latest revision, 8 unresolved discussions.

---

*[BUILD.bazel, line 40 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqEJRQX8Fkq-lQdHmd:-KnqEJRRWRGaUG4YyGBb:bx8gbey) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/BUILD.bazel#L40)):*
> ```
>         "//drake:install",
>         "//drake/common:install",
>         "//drake/lcmtypes:install",
> ```

FYI, I believe this needs to be rebased post-#6401

---

*[tools/install.bzl, line 611 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqEjb1B_QO5M-IXjSH:-KnqEjb1B_QO5M-IXjSI:bnnp73q) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L611)):*
> ```Python
>     jars,
>     dest = "share/java",
>     prefix = "lcmtypes_",
> ```

This is just *strange* for this not being a function in `lcm.bzl`. Rather than defaulting this, since this is only used in one(?) place, can we just move it there instead?

---

*[tools/install.bzl, line 614 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqEyxLpAoSFS4jrtW9:-KnqEyxLpAoSFS4jrtWA:b-a7k14w) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L614)):*
> ```Python
>     suffix = "_java",
> ):
>     """Create a dictionary based on the list of files and targets
> ```

This doesn't really describe *why* you would use this. Suggest instead: "Create an install rename mapping for a set of Java targets." (Also, missing period.)

---

*[tools/install.bzl, line 622 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqFUGCQXlWvQVKf1XN:-KnqFUGDg91i1quBZ7dO:bgii54k) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L622)):*
> ```Python
>     If the element is a target name, the corresponding file name is computed
>     using ``dest`` (which should match the installation folder given to
>     :func:`install`) to which is added the harcoded prefix "lib" and the
> ```

If it's a target, you should be able to retrieve the... actual... ack, that's right, this isn't a rule. Stupid Bazel :cry:. (I'm not sure, though; are we definitely unable to look up labels in this phase?)

---

*[tools/install.bzl, line 628 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqFi_49VdX9EAluvBS:-KnqFi_49VdX9EAluvBT:byhvkz0) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L628)):*
> ```Python
>     If the element is a file name, the new file name is computed by replacing
>     ``dest`` and "lib" with ``prefix`` on the left of the file name and
>     removing ``suffix`` from the right.
> ```

This is confusing. Instead, please make the parameters:

- `strip_prefix` (remove from beginning of file-name, if present; default = `"lib"`)
- `strip_suffix` (remove from end of file-name-without-extension if present; default = `"_java"`)
- `new_name_pattern` (used as `.format()` template to create new name... also, no default for this)

---

*[tools/install.bzl, line 653 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqHmpTzl36w-uiPmQH:-KnqHmpTzl36w-uiPmQI:b-3vsr5y) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L653)):*
> ```Python
>         else:  # File name given
>             if not j.startswith(dest):
>                 fail("{} folder is not {}".format(j, dest))
> ```

Is there a reason for this restriction? We have to guess if a target is given, but why if a file name is given?

---

*[tools/install.bzl, line 654 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqHa6rxS5ZkDaAZCpF:-KnqHa6syZrgeyaqL6QB:b-f2oz3w) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L654)):*
> ```Python
>             if not j.startswith(dest):
>                 fail("{} folder is not {}".format(j, dest))
>             basename = j.rsplit("/", 1)[1]
> ```

There is a function for this in `pathutils.bzl`...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-KnqHEwwIqD2rJI42rjm:b8gq84n)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 0 of 3 files reviewed at latest revision, 8 unresolved discussions.

---

*[BUILD.bazel, line 40 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqEJRQX8Fkq-lQdHmd:-KntK2EshO1pxV3WUHZJ:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/BUILD.bazel#L40)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

FYI, I believe this needs to be rebased post-#6401
</blockquote></details>

Done.

---

*[tools/install.bzl, line 611 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqEjb1B_QO5M-IXjSH:-Knu0zLcpwtrZtSXRxfA:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L611)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

This is just *strange* for this not being a function in `lcm.bzl`. Rather than defaulting this, since this is only used in one(?) place, can we just move it there instead?
</blockquote></details>

Done.

---

*[tools/install.bzl, line 614 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqEyxLpAoSFS4jrtW9:-Knu0zvrZz1O0INncUVL:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L614)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

This doesn't really describe *why* you would use this. Suggest instead: "Create an install rename mapping for a set of Java targets." (Also, missing period.)
</blockquote></details>

Done.

---

*[tools/install.bzl, line 622 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqFUGCQXlWvQVKf1XN:-Knu1-ErPPH4TQ4qtN6s:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L622)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

If it's a target, you should be able to retrieve the... actual... ack, that's right, this isn't a rule. Stupid Bazel :cry:. (I'm not sure, though; are we definitely unable to look up labels in this phase?)
</blockquote></details>

Done.

---

*[tools/install.bzl, line 628 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqFi_49VdX9EAluvBS:-Knu1-hVQNExdsjLJQrI:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L628)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

This is confusing. Instead, please make the parameters:

- `strip_prefix` (remove from beginning of file-name, if present; default = `"lib"`)
- `strip_suffix` (remove from end of file-name-without-extension if present; default = `"_java"`)
- `new_name_pattern` (used as `.format()` template to create new name... also, no default for this)
</blockquote></details>

Done.

---

*[tools/install.bzl, line 653 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqHmpTzl36w-uiPmQH:-Knu10AQCn57B88iswjE:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L653)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Is there a reason for this restriction? We have to guess if a target is given, but why if a file name is given?
</blockquote></details>

Done.

---

*[tools/install.bzl, line 654 at r2](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnqHa6rxS5ZkDaAZCpF:-Knu10cUHGGqeeyD-r4A:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/be6682562668b27094fcdf3a9e1a8089b51ba530/tools/install.bzl#L654)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

There is a function for this in `pathutils.bzl`...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> with some non-blocking nits

---

Reviewed 2 of 2 files at r4.
Review status: 2 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 275 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-Knu1jcTBuhzvfHnu7xc:-Knu1jcTBuhzvfHnu7xd:b-rzphj) ([raw file](https://github.com/robotlocomotion/drake/blob/02efc7823642429fc0c423425419b50fe86242c8/drake/lcmtypes/BUILD#L275)):*
> ```Python
> )
> 
> drake_lcmtypes = [
> ```

BTW, should this be `DRAKE_LCMTYPES`? Or do we have precedent for also non-upper-case variables of this sort? (Similar with others.)

FYI (so other reviewers don't assume I did otherwise) I take your word for it this list is reasonable

---

*[drake/lcmtypes/BUILD, line 310 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-Knu2e-UqDnyk4WBeYDQ:-Knu2e-UqDnyk4WBeYDR:b89w92t) ([raw file](https://github.com/robotlocomotion/drake/blob/02efc7823642429fc0c423425419b50fe86242c8/drake/lcmtypes/BUILD#L310)):*
> ```Python
> drake_java_lcmtypes = [t + "_java" for t in drake_lcmtypes]
> 
> drake_java_lcmtypes_renamed = rename_files(
> ```

BTW, would `DRAKE_JAVA_LCMTYPES_RENAME_MAP` be clearer?

---

*[drake/lcmtypes/BUILD, line 317 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-Knu2N1fWb9GeXsYiNFi:-Knu2N1fWb9GeXsYiNFj:b-bfyf9z) ([raw file](https://github.com/robotlocomotion/drake/blob/02efc7823642429fc0c423425419b50fe86242c8/drake/lcmtypes/BUILD#L317)):*
> ```Python
> )
> 
> # This rule should list every LCM type that is known to Drake or its external
> ```

BTW, this comment might be more appropriately moved to keep it with The Big List...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-Knu3HfLiwa-_MxXHWhO:bki436s)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 1 of 3 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 3 files at r3, 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-Knu3YT2oF_ADocMYrwp:bnfp4nl)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 2 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 275 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-Knu1jcTBuhzvfHnu7xc:-Knu6UzRpGXdquMSP_m8:bf71a55) ([raw file](https://github.com/robotlocomotion/drake/blob/02efc7823642429fc0c423425419b50fe86242c8/drake/lcmtypes/BUILD#L275)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, should this be `DRAKE_LCMTYPES`? Or do we have precedent for also non-upper-case variables of this sort? (Similar with others.)

FYI (so other reviewers don't assume I did otherwise) I take your word for it this list is reasonable
</blockquote></details>

For the list, I listed all the targets that are generated with `lcm_libraries()`. Hopefully I haven't missed one. If I did, at some point we'll realize that and add it then.

---

*[drake/lcmtypes/BUILD, line 310 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-Knu2e-UqDnyk4WBeYDQ:-Knu6pDvTXOhcHAYqOfg:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/02efc7823642429fc0c423425419b50fe86242c8/drake/lcmtypes/BUILD#L310)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, would `DRAKE_JAVA_LCMTYPES_RENAME_MAP` be clearer?
</blockquote></details>

Done.

---

*[drake/lcmtypes/BUILD, line 317 at r4](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-Knu2N1fWb9GeXsYiNFi:-Knu6qHG_TK7QHyJfqXK:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/02efc7823642429fc0c423425419b50fe86242c8/drake/lcmtypes/BUILD#L317)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, this comment might be more appropriately moved to keep it with The Big List...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
+@jwnimmer-tri platform review please

---

Review status: 2 of 3 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-Knu70rqdFtN-vGpA1Cm:b-k0crfs)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 1 of 3 files at r3, 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 276 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuPrGasIznb2Joa8W0:-KnuPrGasIznb2Joa8W1:b-d73l99) ([raw file](https://github.com/robotlocomotion/drake/blob/0254a613cf7fefcf99623b69276c9a50ec8a6921/drake/lcmtypes/BUILD#L276)):*
> ```Python
> 
> # This should list every LCM type that is known to Drake or its external
> # dependencies.
> ```

The "or its external dependencies" part here is a lie.

---

*[drake/lcmtypes/BUILD, line 319 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuPkRsuBLZdsTDN0wQ:-KnuPkRsuBLZdsTDN0wR:b-vh1ozw) ([raw file](https://github.com/robotlocomotion/drake/blob/0254a613cf7fefcf99623b69276c9a50ec8a6921/drake/lcmtypes/BUILD#L319)):*
> ```Python
> )
> 
> java_binary(
> ```

The comment "This should list every LCM type that is known to Drake or its external dependencies." was very important, and must remain on the `drake-lcm-spy` rule here.

---

*[drake/lcmtypes/BUILD, line 336 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuQbobjhD-Mm43moJ8:-KnuQbockMujtOF_wKZQ:b-tpu4vb) ([raw file](https://github.com/robotlocomotion/drake/blob/0254a613cf7fefcf99623b69276c9a50ec8a6921/drake/lcmtypes/BUILD#L336)):*
> ```Python
>     hdr_dest = "include/lcmtypes",
>     rename = DRAKE_JAVA_LCMTYPES_RENAME_MAP,
>     targets = DRAKE_LCMTYPES + DRAKE_JAVA_LCMTYPES,
> ```

BTW I'm unclear on why `DRAKE_LCMTYPES` needs to be here.  Wouldn't `//drake:install` already be installing all of those headers as part of its list of `_LIBDRAKE_COMPONENTS`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 2 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 276 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuPrGasIznb2Joa8W0:-KnuTSr8Z--JF-Yoyl3h:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/0254a613cf7fefcf99623b69276c9a50ec8a6921/drake/lcmtypes/BUILD#L276)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

The "or its external dependencies" part here is a lie.
</blockquote></details>

Done.

---

*[drake/lcmtypes/BUILD, line 319 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuPkRsuBLZdsTDN0wQ:-KnuTTNb2krpAFp57cIq:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/0254a613cf7fefcf99623b69276c9a50ec8a6921/drake/lcmtypes/BUILD#L319)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

The comment "This should list every LCM type that is known to Drake or its external dependencies." was very important, and must remain on the `drake-lcm-spy` rule here.
</blockquote></details>

Done.

---

*[drake/lcmtypes/BUILD, line 336 at r5](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuQbobjhD-Mm43moJ8:-KnuViPy1h0HuaYG4Ii1:b-xqdoew) ([raw file](https://github.com/robotlocomotion/drake/blob/0254a613cf7fefcf99623b69276c9a50ec8a6921/drake/lcmtypes/BUILD#L336)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

BTW I'm unclear on why `DRAKE_LCMTYPES` needs to be here.  Wouldn't `//drake:install` already be installing all of those headers as part of its list of `_LIBDRAKE_COMPONENTS`?
</blockquote></details>

I started this PR before the everything was added to _LIBDRAKE_COMPONENT. It is indeed not necessary anymore.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r6.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-KnuY3zT8wG-R-CZMxzZ:bnfp4nl)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[drake/lcmtypes/BUILD, line 334 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuaS9HRN9YUUxjQR0j:-KnuaS9HRN9YUUxjQR0k:biv6i55) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L334)):*
> ```Python
> install(
>     name = "install",
>     guess_hdrs = "PACKAGE",
> ```

BTW presumably this can also go away...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 333 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuacuZ2tnP1sK3j4a4:-Knuacu_DJE5w2wG6IKI:b-zcunfn) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L333)):*
> ```Python
> 
> install(
>     name = "install",
> ```

BTW, it is a little surprising that this 'is the install rule for this directory', but doesn't install everything. Maybe it should be named `install_java`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 333 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuacuZ2tnP1sK3j4a4:-Knub1UHm2HOBe6wMGOo:bh9kjeh) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L333)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, it is a little surprising that this 'is the install rule for this directory', but doesn't install everything. Maybe it should be named `install_java`?
</blockquote></details>

BTW, or it should install everything....

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 333 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuacuZ2tnP1sK3j4a4:-Knuc5JvF6RRB4doXtTn:b-hkhpyb) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L333)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

BTW, or it should install everything....
</blockquote></details>

Well, yes, but that would revert the change @jwnimmer-tri just requested :smile:. (Or is that TRTTD?)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 333 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuacuZ2tnP1sK3j4a4:-KnueeZjhkqgatqL6-AQ:bc79puu) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L333)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

Well, yes, but that would revert the change @jwnimmer-tri just requested :smile:. (Or is that TRTTD?)
</blockquote></details>

The LCM types are somewhat of a special case.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Can you also install the Python in this PR?

---

Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-KoI3g8F3K1lvEd4w5UQ:b-7al69j)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 2 of 5 files reviewed at latest revision, 2 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 333 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuacuZ2tnP1sK3j4a4:-KoJAoX7gwxcf_SNjB1J:b-tkngz6) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L333)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

The LCM types are somewhat of a special case.
</blockquote></details>

I have moved the installation of the drake lcmtype headers back into this install rule, and added an interface target for them in drake.cps. I also added the java targets in drake.cps. Let me know if that is what you had in mind.

---

*[drake/lcmtypes/BUILD, line 334 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuaS9HRN9YUUxjQR0j:-KoJBDj8QCrhlI5KERej:b-emtfc8) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L334)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW presumably this can also go away...
</blockquote></details>

Actually the `guess_hdrs` is required for all the headers such as `lcmt_acrobot_u.hpp` to install

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 5 of 5 files at r7.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/BUILD, line 16 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCETmpcU33WPhjXd2:-KoJCETmpcU33WPhjXd3:b-ramy49) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/drake/BUILD#L16)):*
> ```Python
> # bazel query 'kind("cc_library", visible("//drake:libdrake.so", "//drake/..."))
> # except("//drake/examples/...")
> # except(attr("testonly", "1", "//drake/..."))'
> ```

Now that `lcmtypes` have been removed this comment no longer matches the below list, I think?

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoJCYEM9OLhsB1Do8MC:bdq122j) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
> ```delphi
>       "Type": "jar",
>       "Location": "@prefix@/share/java/lcmtypes_zmp_com_observer_state.jar",
>       "Requires": ["lcm:lcm-java"]
> ```

This list of jarfiles seems so long as to be completely unmaintainable.  At least add a TODO to make it less brittle, or fail-fast, or be generated, or something?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>, though per comments, it would be nice to do something better for the CPS...

---

Reviewed 3 of 5 files at r7.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/lcmtypes/BUILD, line 334 at r6](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KnuaS9HRN9YUUxjQR0j:-KoKncCuLWZCgwBxpKdU:b3azypy) ([raw file](https://github.com/robotlocomotion/drake/blob/b08d80e168d8d201ed2bab51f5de84cdfd189eda/drake/lcmtypes/BUILD#L334)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

Actually the `guess_hdrs` is required for all the headers such as `lcmt_acrobot_u.hpp` to install
</blockquote></details>

Well, yes, now that we're installing the C++ bits again :smile:.

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoKp1IWKTDN0I858_uV:b-z7erit) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

This list of jarfiles seems so long as to be completely unmaintainable.  At least add a TODO to make it less brittle, or fail-fast, or be generated, or something?
</blockquote></details>

I was thinking the same thing. If these are made not-last (so that there is not a trailing-comma issue), and we switch to a `drake-create-cps.py`, this could possibly be refactored to something like:

```python
java_component_template = """
    "%s-lcmtypes-java": {
      "Type": "jar",
      "Location": "@prefix@/share/java/lcmtypes_%s.jar",
      "Requires": ["lcm:lcm-java"]
    },
"""[1:] # remove leading newline
java_targets = [ ... ]
java_components = sum([java_component_template % (t, t) for t in java_targets])

content = """
...other CPS contents...
%s
...other CPS contents...
""" % java_components[:-1]
```

...and possibly we could also arrange somehow to pass the list of targets into the script.

(A better solution is to teach Bazel to generate CPS more directly. However, this was my original plan way back when, and after considering it for a while, I did not come up with a practical way of attaining this ideal.)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-KoKrW0Pgf9bg-GgFL4b:bn2iv9p)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoMp-XT5CpdG417vA3z:b-iz9h76) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

I was thinking the same thing. If these are made not-last (so that there is not a trailing-comma issue), and we switch to a `drake-create-cps.py`, this could possibly be refactored to something like:

```python
java_component_template = """
    "%s-lcmtypes-java": {
      "Type": "jar",
      "Location": "@prefix@/share/java/lcmtypes_%s.jar",
      "Requires": ["lcm:lcm-java"]
    },
"""[1:] # remove leading newline
java_targets = [ ... ]
java_components = sum([java_component_template % (t, t) for t in java_targets])

content = """
...other CPS contents...
%s
...other CPS contents...
""" % java_components[:-1]
```

...and possibly we could also arrange somehow to pass the list of targets into the script.

(A better solution is to teach Bazel to generate CPS more directly. However, this was my original plan way back when, and after considering it for a while, I did not come up with a practical way of attaining this ideal.)
</blockquote></details>

Another solution would be to put all of `drake/lcmtypes` into a single jar file.  I no longer see any reason to try to keep them as separate jars, even though I think it was me who started that trend.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN-M6qx0VbUU9OSNfa:b-xpn3ly) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Another solution would be to put all of `drake/lcmtypes` into a single jar file.  I no longer see any reason to try to keep them as separate jars, even though I think it was me who started that trend.
</blockquote></details>

@mwoehlke: I am working on a solution close to that. I'll compare both solutions and update my code.
@jwnimmer-tri: Is there a way with bazel to merge multiple jars into one jar? I tried with `java_library()` and `java_binary()`, but I haven't found a way. Otherwise I can also call `lcm_libraries()` with all the source files at once, that is the easy solution.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Reviewed 5 of 5 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN0OSTw_bazFYnlexH:b-1e2dif) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

@mwoehlke: I am working on a solution close to that. I'll compare both solutions and update my code.
@jwnimmer-tri: Is there a way with bazel to merge multiple jars into one jar? I tried with `java_library()` and `java_binary()`, but I haven't found a way. Otherwise I can also call `lcm_libraries()` with all the source files at once, that is the easy solution.
</blockquote></details>

Yes, but I am not sure we even need separate jars.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN0Vn7uqIhNRJ6CWxM:b7kllzn) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

Yes, but I am not sure we even need separate jars.
</blockquote></details>

(I think @jwnimmer-tri still wants multiple C++ libraries)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN0mY2OLRrAIPMZiNU:b-ijyylr) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
> I can also call lcm_libraries() with all the source files at once, that is the easy solution.

Yes -- I meant feed in all of the `*.lcm` files at once to get a single `jar` right out of the gate, not combining them all later.  So like we do with python.

> I think @jwnimmer-tri still wants multiple C++ libraries.

Yes please.  I don't know why `lcm_libraries` needs to do both C++ and Java at once, because I don't know why we'd ever want separate jars.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN1REQBGXvgeNAZixE:b-hs9wkm) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
> Yes please. I don't know why lcm_libraries needs to do both C++ and Java at once, because I don't know why we'd ever want separate jars.

It was because you were planning on relocating the LCM files into different Bazel packages.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN25DSW0X6mdpv4r8j:b-ms0lol) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

> Yes please. I don't know why lcm_libraries needs to do both C++ and Java at once, because I don't know why we'd ever want separate jars.

It was because you were planning on relocating the LCM files into different Bazel packages.
</blockquote></details>

Sure, but they will still be in rough groups (packages), so we'd have one jar per package, not one jar per message.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN2QjFJXbO7GyWgpod:b-9zdwja) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Sure, but they will still be in rough groups (packages), so we'd have one jar per package, not one jar per message.
</blockquote></details>

I would argue the C++ should be the same given how little you gain since they are header-only, but that is an argument for another time and place.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
*[tools/drake.cps, line 203 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCYELL3WTALZ-89zX:-KoN4HlLQc0j0lhpBojc:bmu81s6) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/drake.cps#L203)):*
<details><summary><i>Previously, jamiesnape (Jamie Snape) wrote…</i></summary><blockquote>

I would argue the C++ should be the same given how little you gain since they are header-only, but that is an argument for another time and place.
</blockquote></details>

Once they are in a few separate packages, I think I'll agree.  What I want to avoid is the license clusterfrag that is the libbot ecosystem polluting the dependency tree of e.g. automotive.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 1 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

*[drake/BUILD, line 16 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCETmpcU33WPhjXd2:-KoOC4AGkx3WJEfFm6fT:b-s7gj4q) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/drake/BUILD#L16)):*
<details><summary><i>Previously, jwnimmer-tri (Jeremy Nimmer) wrote…</i></summary><blockquote>

Now that `lcmtypes` have been removed this comment no longer matches the below list, I think?
</blockquote></details>

I have updated the query and the list accordingly.
I am not sure though if "//drake:gurobi_deps", "//drake:mosek_deps", or "//drake_vtk_deps" should be in the list, or filtered out.

---

*[tools/install.bzl, line 591 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoOCTBRjNpGoHtL-N9O:-KoOCTBRjNpGoHtL-N9P:b-4lsnu7) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/install.bzl#L591)):*
> ```Python
>         visibility = visibility,
>     )
> ```

@mwoehlke: We actually don't need this anymore :( I left your commit in the history so it is easy to get it back if we need it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: 1 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/install.bzl, line 591 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoOCTBRjNpGoHtL-N9O:-KoOEFmQkJC5kdSFe0is:brs28w8) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/install.bzl#L591)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

@mwoehlke: We actually don't need this anymore :( I left your commit in the history so it is easy to get it back if we need it.
</blockquote></details>

I'd prefer to remove the commit entirely and instead make it available on a branch somewhere. (Also, your revert is incomplete, and you lost a blank line that is supposed to be present.)

Personally, I'd be okay just leaving the function even if it is not used...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 1 of 6 files reviewed at latest revision, 3 unresolved discussions.

---

*[tools/install.bzl, line 591 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoOCTBRjNpGoHtL-N9O:-KoOLEMdlck62FG3aHLy:bm7e5wm) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/install.bzl#L591)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

I'd prefer to remove the commit entirely and instead make it available on a branch somewhere. (Also, your revert is incomplete, and you lost a blank line that is supposed to be present.)

Personally, I'd be okay just leaving the function even if it is not used...
</blockquote></details>

I removed it (less code to maintain). I'll keep a copy of the modification in a branch in my repo.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 6 of 6 files at r8.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[drake/BUILD, line 16 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoJCETmpcU33WPhjXd2:-KoOSCNg071V9saEFGKw:bdq9x0o) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/drake/BUILD#L16)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

I have updated the query and the list accordingly.
I am not sure though if "//drake:gurobi_deps", "//drake:mosek_deps", or "//drake_vtk_deps" should be in the list, or filtered out.
</blockquote></details>

I think if `:drake_shared_library` is excluded then also its immediate `deps=` should also be excluded (such as `:mosek_deps`) since their only purpose is to be used by `:drake_shared_library`.

However, I don't think we need to worry about fine-tuning this list yet, as long as the comment and output are consistent.  Once #6464 gets underway we can do a more careful review of what's included.  Perhaps this discussion can just be added to #6464 as one more TODO.

I've marked myself satisfied here.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-:-KoOWJCZEafRvDT1Vt24:bnfp4nl)*
<!-- Sent from Reviewable.io -->

mwoehlke-kitware commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/lcmtypes/BUILD, line 310 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoOW_1-1P3MoFgnIk8c:-KoOW_1-1P3MoFgnIk8d:bxh750x) ([raw file](https://github.com/robotlocomotion/drake/blob/b3f7a8137d9eae417667851f704c2e8f0b885765/drake/lcmtypes/BUILD#L310)):*
> ```Python
> install(
>     name = "install",
>     targets = DRAKE_LCMTYPES + [":lcmtypes_drake_java"] + [":lcmtypes-py"],
> ```

BTW, two lists? Probably one list with two items is better :smile:...

---

*[tools/install.bzl, line 591 at r7](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoOCTBRjNpGoHtL-N9O:-KoOWxd4an0mIdydZGol:b-78dy36) ([raw file](https://github.com/robotlocomotion/drake/blob/7d530035bad4b658b3ec0bbf32ee76ecc9a5d723/tools/install.bzl#L591)):*
<details><summary><i>Previously, fbudin69500 (Francois Budin) wrote…</i></summary><blockquote>

I removed it (less code to maintain). I'll keep a copy of the modification in a branch in my repo.
</blockquote></details>

:+1: (I believe it is in my fork somewhere as well, but more copies are good...)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

fbudin69500 commented:
Review status: 4 of 6 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/lcmtypes/BUILD, line 310 at r8](https://reviewable.io:443/reviews/robotlocomotion/drake/6477#-KoOW_1-1P3MoFgnIk8c:-KoOj5peUW7Rb4D-b6M9:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/b3f7a8137d9eae417667851f704c2e8f0b885765/drake/lcmtypes/BUILD#L310)):*
<details><summary><i>Previously, mwoehlke-kitware (Matthew Woehlke) wrote…</i></summary><blockquote>

BTW, two lists? Probably one list with two items is better :smile:...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jamiesnape commented:
Reviewed 2 of 2 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->

jwnimmer-tri commented:
Reviewed 2 of 2 files at r9.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/6477)*
<!-- Sent from Reviewable.io -->
